src/terms/ewald.jl

```diff
@@ -187,19 +186,24 @@ function energy_ewald(lattice, recip_lattice, charges, positions; η=nothing, fo
                     continue
                 end
 
-                dist = norm(lattice * (ti - tj - R))
+                v = lattice * (ti - tj - R)
+                dist = norm(v)
 
                 # erfc decays very quickly, so cut off at some point
                 if η * dist > max_erfc_arg
                     continue
                 end
 
                 any_term_contributes = true
-                sum_real += Zi * Zj * erfc(η * dist) / dist
+                energy_contribution = Zi * Zj * erfc(η * dist) / dist 
+                sum_real += energy_contribution
                 if forces !== nothing
-                    # Use ForwardDiff here because I'm lazy. TODO do it properly
-                    forces_real[i] -= ForwardDiff.gradient(r -> (dist=norm(lattice * (r - tj - R)); Zi * Zj * erfc(η * dist) / dist), ti)
-                    forces_real[j] -= ForwardDiff.gradient(r -> (dist=norm(lattice * (ti - r - R)); Zi * Zj * erfc(η * dist) / dist), tj)
+                    # grad = ForwardDiff.gradient(r -> (dist=norm(lattice * (r - tj - R)); Zi * Zj * erfc(η * dist) / dist), ti)
+                    ddist = Zi * Zj * η * (-2exp(-(η*dist)^2) / sqrt(T(π)))
+                    ddist += -energy_contribution
+                    dti = lattice'*((ddist / dist^2) * v)
+                    forces_real[i] += -dti
```

Comment:
-= also here

Reply:
Or is it on purpose because you find it more readable? I don't have a strong opinion so if you prefer it like that leave it.

Reply:
During my derivation it was on purpose, but I agree it's not necessary.